packages/server/src/messages/api.ts

```diff
@@ -1,178 +1,156 @@
-        const { error } = Schema.Api.Turn.validate(req.params)
-        if (error) {
-          return res.status(400).send(error.message)
-        }
+    res.send({ message, responses: await collector })
+  }
 
-        const { id } = req.params
+  async get(req: ClientApiRequest, res: Response) {
+    const id = req.params.id as uuid
 
-        const message = await this.messages.get(id)
-        if (!message) {
-          return res.sendStatus(404)
-        }
+    const message = await this.messages.get(id)
+    if (!message) {
+      return res.send(undefined)
+    }
 
-        const conversation = await this.conversations.get(message.conversationId)
-        if (!conversation) {
-          return res.sendStatus(404)
-        } else if (conversation.clientId !== req.client!.id) {
-          return res.sendStatus(403)
-        }
+    const conversation = await this.conversations.get(message.conversationId)
+    if (!conversation || conversation.clientId !== req.client.id) {
+      return res.send(undefined)
+    }
 
-        await this.converse.stopCollecting(message.id, message.conversationId)
-        res.sendStatus(200)
-      })
-    )
+    res.send(message)
+  }
 
-    this.router.get(
-      '/messages/:id',
-      this.auth.client.auth(async (req, res) => {
-        const { error } = Schema.Api.Get.validate(req.params)
-        if (error) {
-          return res.status(400).send(error.message)
-        }
+  async list(req: ClientApiRequest, res: Response) {
+    const conversationId = req.params.conversationId as uuid
+    const limit = +(req.query.limit || 20)
 
-        const { id } = req.params
-        const message = await this.messages.get(id)
+    const conversation = await this.conversations.get(conversationId)
+    if (!conversation || conversation.clientId !== req.client.id) {
+      return res.sendStatus(404)
+    }
 
-        if (!message) {
-          return res.sendStatus(404)
-        }
+    const messages = await this.messages.listByConversationId(conversationId, limit)
+    res.send(messages)
+  }
 
-        const conversation = await this.conversations.get(message.conversationId)
-        if (conversation!.clientId !== req.client!.id) {
-          return res.sendStatus(403)
-        }
+  async delete(req: ClientApiRequest, res: Response) {
+    const { id } = req.params
 
-        res.send(message)
-      })
-    )
+    const message = await this.messages.get(id)
+    if (!message) {
+      return res.send(false)
+    }
 
-    this.router.get(
-      '/messages',
-      this.auth.client.auth(async (req, res) => {
-        const { error } = Schema.Api.List.validate(req.query)
-        if (error) {
-          return res.status(400).send(error.message)
-        }
+    const conversation = await this.conversations.get(message.conversationId)
+    if (!conversation || conversation.clientId !== req.client.id) {
+      return res.send(false)
+    }
+
+    await this.messages.delete(id)
+    res.send(true)
```

Comment:
Usually with delete call, a 204 no content means that the resource was properly deleted.

Reply:
For this route we consider it valid to try to delete a ressource that doesn't exist. What should we return when the ressource doesn't exist? If we returned an error code it would mean that the route doesn't accept trying to delete something that is already deleted.

Reply:
I might be mistaken, but if I recall correctly, the standard is to always return 204 no matter what (maybe 404 too, not sure).

Look at the second answer from this thread, I think it makes sense: https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time